### PR TITLE
Project: fetch project data on the server

### PR DESCRIPTION
**Pull request by eatyourgreens**
Remove `project.fetch` from the Project model. Replace it with `helpers/fetchProjectData`, which is imported into `getStaticPageProps` and only runs on the server. That should remove unused API code from client-side code bundles.

> Note: You can import modules in top-level scope for use in getServerSideProps. Imports used in getServerSideProps will not be bundled for the client-side.
>
>This means you can write server-side code directly in getServerSideProps. This includes reading from the filesystem or a database.

https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering

Package:
app-project

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

File: packages/app-project/src/helpers/getDefaultPageProps/getDefaultPageProps.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default async function getDefaultPageProps({ params, query, req, res }) {
   const dismissedAnnouncementBanner = getCookie(req, 'dismissedAnnouncementBanner') || null
 
   const { props: staticProps } = await getStaticPageProps({ params, query })
-  const { project, isServer, statusCode, title, workflowID, workflows } = staticProps
+  const { project, statusCode, title, workflowID, workflows } = staticProps
```
Comment (eatyourgreens) [inline packages/app-project/src/helpers/getDefaultPageProps/getDefaultPageProps.js:11]: `isServer` wasn't being used anywhere in this file. I think it's left over from `getInitialProps`, which runs on the server and also in the browser.

**Review by goplayoutside3 — APPROVED** [review]

**Comment by goplayoutside3** [comment]
@eatyourgreens once this branch was merged to `master`, I can no longer load my test project without several errors about 'active workflows'. Visiting [http://localhost:3000/projects/brooke/i-fancy-cats](http://localhost:3000/projects/brooke/i-fancy-cats) works as expected on `master`, but I can't use my test project anymore [http://localhost:3000/projects/goplayoutside3/test-bike-lane-uprising](http://localhost:3000/projects/goplayoutside3/test-bike-lane-uprising).

I see an error `Cannot read property 'length' of undefined` from this function:
https://github.com/zooniverse/front-end-monorepo/blob/386da9ee5ecd504ae34610ad394ac874b2afe39d/packages/app-project/stores/Project.js#L40-L47

It seems active workflows are no longer linked, but I don't understand enough about what changed in branch `project-fetch` to find why the above error shows for my project, but not i-fancy-cats. My test project does have a workflow.

**Comment by eatyourgreens** [comment]
Does adding a researcher quote to your project (in the project builder)fix it? I think I know what the problem is.

**Comment by goplayoutside3** [comment]
Yes you're right! I was just going to comment that it has nothing to do with active workflows haha.

**Comment by eatyourgreens** [comment]
@goplayoutside3 #2053 should fix it. I'll add a link to a project with a null researcher quote to that PR.

**Comment by eatyourgreens** [comment]
Sorry. That was my fault. I think the underlying reason is that `undefined` is invalid JSON, so empty values are serialised as null. 

You may run into the inverse problem while adding new models to the store. If a model contains undefined values, Next will error and warn that the page props can’t be serialised to JSON. If you ever run into that error, setting null as a default for undefined properties can fix it.

**Pull request merged**